src/controller/artifact/processor/model/model.go

```diff
@@ -0,0 +1,106 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"context"
+	"encoding/json"
+
+	modelspec "github.com/CloudNativeAI/model-spec/specs-go/v1"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+
+	ps "github.com/goharbor/harbor/src/controller/artifact/processor"
+	"github.com/goharbor/harbor/src/controller/artifact/processor/base"
+	"github.com/goharbor/harbor/src/controller/artifact/processor/model/parser"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/pkg/artifact"
+)
+
+// const definitions
+const (
+	// ArtifactTypeModel defines the artifact type for AI model.
+	ArtifactTypeModel = "MODEL"
```

Comment:
I also have this here https://github.com/goharbor/harbor/blob/01b4b23856ce1e1fb9a931c763ecaa040f366ef7/src/controller/artifact/processor/cnai/cnai.go#L36 and I feel like we should consider changing `MODEL` to something like `CNAI`, Harbor can possibly support multiple ML models in future

Reply:
+1 to update it to `CNAI`